### PR TITLE
Add Makefile and Circle CI config

### LINKED ISSUE #13: Enable misspell go tool
Please enable the tool you enabled on specification here as well. Can you please use circleCi for that?

### PR DESCRIPTION
**Pull request by songy23**
Fixes #13.
Currently the only job is misspell check.

### CODE REVIEW

File: .circleci/config.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+version: 2
```
Comment (songy23) [inline .circleci/config.yaml]: I'm less familiar with Circle CI setup so gently paging @rghetia and @pavolloffay for a review.

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+	$(MISSPELL) $(ALL_DOC)
+
+.PHONY: misspell-correction
+misspell-correction:
+	$(MISSPELL_CORRECTION) $(ALL_DOC)
+
+.PHONY: install-tools
+install-tools:
+	GO111MODULE=on go install \
+	  github.com/client9/misspell/cmd/misspell
```
Comment (rghetia) [inline Makefile]: Use this [approach](https://github.com/open-telemetry/opentelemetry-go/blob/623b062e17fb1aebae74aad74d1fc637bde520b2/Makefile#L19-L20) to install tools locally to guarantee that same version is used.

Comment (songy23) [reply Makefile]: Thanks, updated in https://github.com/open-telemetry/rfcs/pull/14/commits/1d30f395916a8c5070c091e1a2df6b07901cd389.

**Comment by rghetia** [comment]
@songy23 the failure may be due to filename  .circleci.yaml instead of .circleci.yml

**Comment by songy23** [comment]
@rghetia That fixed it - thanks!

**Pull request merged**